lib/fs/filesystem.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package fs
+
+import (
+	"io"
+	"os"
+	"time"
+)
+
+type LinkTargetType int
+
+const (
+	LinkTargetFile LinkTargetType = iota
+	LinkTargetDirectory
+	LinkTargetUnknown
+)
+
+// The Filesystem interface abstracts access to the file system.
+type Filesystem interface {
+	ChangeSymlinkType(name string, tt LinkTargetType) error
+	Chmod(name string, mode os.FileMode) error
```

Comment:
I say os.FileMode should be a redefined locally, copying stuff from stdlib or potentially even using an interface.
I know there is no need now, but we might add more bits to it?